File: User_App/android/app/src/main/java/vn/iflux/app/ui/components/IfxComponents.kt
package vn.iflux.app.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import vn.iflux.app.core.model.IfxSector
import vn.iflux.app.core.model.IfxStock
import vn.iflux.app.ui.theme.IfxColors

@Composable
fun IfxBrandName() {
    Text(
        text = "iFlux",
        color = IfxColors.TextPrimary,
        fontSize = 20.sp,
        fontWeight = FontWeight.Bold,
        modifier = Modifier
            .background(IfxColors.BrandPrimary, RoundedCornerShape(6.dp))
            .padding(horizontal = 6.dp, vertical = 2.dp)
    )
}

@Composable
fun IfxStockRow(stock: IfxStock, modifier: Modifier = Modifier) {
    Row(modifier = modifier.fillMaxWidth().padding(vertical = 8.dp), verticalAlignment = Alignment.CenterVertically) {
        Column(Modifier.weight(1f)) {
            Text(stock.ticker, color = IfxColors.TextPrimary, fontWeight = FontWeight.SemiBold)
            stock.name?.let { Text(it, color = IfxColors.TextSecondary, fontSize = 12.sp) }
        }
        Column(horizontalAlignment = Alignment.End) {
            Text(
                stock.price?.let { String.format("%.2f", it) } ?: "—",
                color = IfxColors.TextPrimary,
                fontFamily = FontFamily.Monospace,
                fontWeight = FontWeight.Bold
            )
            val change = stock.changePct ?: 0.0
            Text(
                String.format("%+.2f%%", change),
                color = if (change >= 0) IfxColors.Positive else IfxColors.Negative,
                fontSize = 12.sp,
                fontFamily = FontFamily.Monospace
            )
        }
    }
}

@Composable
fun IfxSectorCard(sector: IfxSector) {
    Column(
        Modifier
            .fillMaxWidth()
            .background(IfxColors.SurfaceElevated, RoundedCornerShape(12.dp))
            .padding(12.dp)
    ) {
        Text(sector.name, color = IfxColors.TextPrimary, fontWeight = FontWeight.SemiBold)
        Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
            Text("Ig ${"%.1f".format(sector.ig)}", color = IfxColors.TextSecondary, fontSize = 13.sp)
            Text(
                String.format("%+.2f%%", sector.pg),
                color = if (sector.pg >= 0) IfxColors.Positive else IfxColors.Negative,
                fontSize = 13.sp,
                fontWeight = FontWeight.SemiBold
            )
        }
    }
}

@Composable
fun IfxPrimaryButton(text: String, onClick: () -> Unit) {
    Button(
        onClick = onClick,
        modifier = Modifier.fillMaxWidth(),
        colors = ButtonDefaults.buttonColors(containerColor = IfxColors.BrandPrimary)
    ) {
        Text(text)
    }
}

File: User_App/ios/iFlux/Design/IfxComponents.swift
import SwiftUI

struct IfxBrandName: View {
    var body: some View {
        Text("iFlux")
            .font(.system(size: 20, weight: .bold))
            .foregroundStyle(IfxTokens.brandOnPrimary)
            .padding(.horizontal, 6)
            .padding(.vertical, 2)
            .background(IfxTokens.brandPrimary)
            .clipShape(RoundedRectangle(cornerRadius: IfxTokens.radiusBrand, style: .continuous))
    }
}

struct IfxStockRow: View {
    let stock: IfxStock

    var body: some View {
        HStack(spacing: IfxTokens.space3) {
            VStack(alignment: .leading, spacing: 2) {
                Text(stock.ticker)
                    .font(.system(size: 15, weight: .semibold))
                    .foregroundStyle(IfxTokens.textPrimary)
                if let name = stock.name {
                    Text(name)
                        .font(.system(size: 12))
                        .foregroundStyle(IfxTokens.textSecondary)
                        .lineLimit(1)
                }
            }
            Spacer()
            VStack(alignment: .trailing, spacing: 2) {
                Text(formatPrice(stock.price))
                    .font(.system(size: 15, weight: .bold, design: .monospaced))
                    .foregroundStyle(IfxTokens.textPrimary)
                Text(formatChange(stock.changePct))
                    .font(.system(size: 12, weight: .semibold, design: .monospaced))
                    .foregroundStyle(changeColor(stock.changePct))
            }
        }
        .padding(.vertical, IfxTokens.space2)
    }

    private func formatPrice(_ value: Double?) -> String {
        guard let value else { return "—" }
        return String(format: "%.2f", value)
    }

    private func formatChange(_ value: Double?) -> String {
        guard let value else { return "—" }
        return String(format: "%+.2f%%", value)
    }

    private func changeColor(_ value: Double?) -> Color {
        guard let value else { return IfxTokens.textSecondary }
        return value >= 0 ? IfxTokens.positive : IfxTokens.negative
    }
}

struct IfxSectorCard: View {
    let sector: IfxSector

    var body: some View {
        VStack(alignment: .leading, spacing: IfxTokens.space2) {
            Text(sector.name ?? "Ngành \(sector.id)")
                .font(.system(size: 14, weight: .semibold))
                .foregroundStyle(IfxTokens.textPrimary)
            HStack {
                Text(String(format: "Ig %.1f", sector.ig))
                    .font(.system(size: 13, design: .monospaced))
                Spacer()
                Text(String(format: "%+.2f%%", sector.pg))
                    .font(.system(size: 13, weight: .semibold, design: .monospaced))
                    .foregroundStyle(sector.pg >= 0 ? IfxTokens.positive : IfxTokens.negative)
            }
            .foregroundStyle(IfxTokens.textSecondary)
        }
        .padding(IfxTokens.space3)
        .background(IfxTokens.surfaceElevated)
        .clipShape(RoundedRectangle(cornerRadius: IfxTokens.radiusMd))
        .overlay(RoundedRectangle(cornerRadius: IfxTokens.radiusMd).stroke(IfxTokens.border))
    }
}

struct IfxPrimaryButton: View {
    let title: String
    var action: () -> Void

    var body: some View {
        Button(action: action) {
            Text(title)
                .font(.system(size: 15, weight: .semibold))
                .frame(maxWidth: .infinity)
                .padding(.vertical, 12)
                .background(IfxTokens.brandPrimary)
                .foregroundStyle(IfxTokens.brandOnPrimary)
                .clipShape(RoundedRectangle(cornerRadius: IfxTokens.radiusSm))
        }
    }
}

struct IfxScreenShell<Content: View>: View {
    let title: String
    @ViewBuilder var content: () -> Content

    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: IfxTokens.space4) {
                Text(title)
                    .font(.system(size: 22, weight: .bold))
                    .foregroundStyle(IfxTokens.textPrimary)
                content()
            }
            .padding(IfxTokens.space4)
        }
        .background(IfxTokens.surface.ignoresSafeArea())
    }
}
